file_transfers.h

```diff
@@ -0,0 +1,57 @@
+#define MAX_FILE_TRANSFERS 32
+
+enum UTOX_FILE_TRANSFER_STATUS{
+    FILE_TRANSFER_STATUS_NONE,
+    FILE_TRANSFER_STATUS_ACTIVE,
+    FILE_TRANSFER_STATUS_PAUSED_US,
+    FILE_TRANSFER_STATUS_PAUSED_BOTH,
+    FILE_TRANSFER_STATUS_PAUSED_THEM,
+    FILE_TRANSFER_STATUS_BROKEN,
+    FILE_TRANSFER_STATUS_COMPLETED,
+    FILE_TRANSFER_STATUS_KILLED,
+};
+
+typedef struct FILE_TRANSFER {
+    uint32_t friend_number, file_number;
+    uint8_t status;
+    _Bool in_memory, incoming;
+    uint8_t *path, *name, *file_id;
+    size_t path_length, name_length, size, size_transferred;
+    uint8_t *memory;
+    FILE *file;
+    time_t request_time, start_time, last_chunk_time, finish_time, pause_time;
+    MSG_FILE *ui_data;
+} FILE_TRANSFER;
+
+typedef struct {
+    uint64_t size_transferred;
+    uint32_t speed;
+} FILE_PROGRESS;
+
+/* Callback for file transfer changes made by friend */
+void file_transfer_local_control(Tox *tox, uint32_t friend_number, uint32_t file_number, TOX_FILE_CONTROL control);
+static void file_transfer_callback_control(Tox *tox, uint32_t friend_number, uint32_t file_number, TOX_FILE_CONTROL control, void *UNUSED(userdata));
```

Comment:
I wanted to put all the internal functions at the bottom of the .c while still using them above. I can re order the functions, and remove the prototypes if you'd like.

I did in this was so I can just write the documentation in the .h and not miss any functions...